### PR TITLE
feat(authn): added a HTTP backend for the authentication mechanism scram

### PR DESCRIPTION
**Pull request by lafirest**
Fixes <issue-or-jira-number>

Release version: v/e5.?

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [ ] Added tests for the changes
- [ ] Added property-based tests for code which performs user input validation
- [ ] Changed lines covered in coverage report
- [ ] Change log has been added to `changes/(ce|ee)/(feat|perf|fix|breaking)-<PR-id>.en.md` files
- [ ] For internal contributor: there is a jira ticket to track this change
- [ ] Created PR to [emqx-docs](https://github.com/emqx/emqx-docs) if documentation update is required, or link to a follow-up jira ticket
- [ ] Schema changes are backward compatible

## Checklist for CI (.github/workflows) changes

- [ ] If changed package build workflow, pass [this action](https://github.com/emqx/emqx/actions/workflows/build_packages.yaml) (manual trigger)
- [ ] Change log has been added to `changes/` dir for user-facing artifacts update

### CODE REVIEW

**Review by HJianBo — APPROVED** [review]
LGTM

File: changes/ee/feat-13504.en.md [hunk 1]
```diff
@@ -0,0 +1 @@
+Added a HTTP backend for the authentication mechanism `scram`.
```
Comment (zmstone) [inline changes/ee/feat-13504.en.md:1]: please briefly describe how it works.
should also emphasize that this is not rfc7804

File: apps/emqx_auth_http/src/emqx_authn_scram_http.erl [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+    destroy/1
+]).
+
+-define(REQUIRED_USER_INFO_KEYS, [
+    <<"stored_key">>,
+    <<"server_key">>,
+    <<"salt">>
+]).
+
+-define(OPTIONAL_USER_INFO_KEYS, [
```
Comment (zmstone) [inline apps/emqx_auth_http/src/emqx_authn_scram_http.erl:25]: A separate (maybe low prio) PR after this is merged to `master`:
add `<<"acl">>` here, also add some tests to verify ACL checks.

Comment (zmstone) [reply apps/emqx_auth_http/src/emqx_authn_scram_http.erl:25]: it's actually already in release-57: https://github.com/emqx/emqx/pull/13400

**Comment by emqxqa** [comment]
[TestExecution](https://emqx.atlassian.net/browse/QA-3227)

**Pull request merged**